core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -344,10 +354,22 @@ public void shutdown() {
     public void setOutOfMessages(java.util.function.Consumer<Consumer> outOfMessages) {
         this.outOfMessages = outOfMessages;
     }
+    
+    private final void configureConstructorInjection() {
+        Constructor<?>[] constructors = msgHandler.getDeclaredConstructors();
+        for (Constructor<?> constructor : constructors) {
+            if (constructor.isAnnotationPresent(Inject.class)) {
+                this.constructor = constructor;
```

Comment:
Maybe worth it to log an error or warning (realistically, logging an error is more immediately useful) when there are multiple constructors annotated with `@Inject`?